File: fixtures/scale/java/medium/src/calc/repl/Session.java
package calc.repl;

import calc.config.Settings;
import calc.eval.Environment;
import calc.eval.FunctionRegistry;
import calc.funcs.BuiltinInstaller;
import calc.history.History;

public class Session {
    private final Environment environment = new Environment();
    private final FunctionRegistry registry = new FunctionRegistry();
    private final History history = new History(200);
    private final Settings settings = new Settings();

    public Session() {
        BuiltinInstaller.install(registry, settings);
    }

    public Environment getEnvironment() {
        return environment;
    }

    public FunctionRegistry getRegistry() {
        return registry;
    }

    public History getHistory() {
        return history;
    }

    public Settings getSettings() {
        return settings;
    }
}

File: fixtures/scale/dart/medium/lib/repl/session.dart
import '../app/installer.dart';
import '../config/settings.dart';
import '../eval/environment.dart';
import '../eval/function_registry.dart';
import '../funcs/constants.dart';
import '../history/history.dart';

class Session {
  final Environment environment = Environment();
  final FunctionRegistry registry = FunctionRegistry();
  final History history = History(200);
  final Settings settings = Settings();

  Session() {
    _bootstrap();
  }

  void _bootstrap() {
    installBuiltins(registry, settings);
    seedConstants(environment);
  }
}
